pysoa/common/transport/redis_gateway/backend/sentinel.py

```diff
@@ -117,7 +135,18 @@ def _get_connection(self, index=None):
                 )
             )
 
-        return self._master_for(self._services[index])
+        for i in range(self._sentinel_failover_retries + 1):
+            try:
+                return self._master_for(self._services[index])
+            except redis.sentinel.MasterNotFoundError:
+                self._last_sentinel_refresh = 0  # make sure we reach out to get master info again on next call
+                self._get_counter('backend.sentinel.master_not_found_retry').increment()
+                time.sleep((1 ** i + random.random()) / 4.0)
```

Comment:
What range are you trying to produce on each iteration?

Won't `1 ** i` always equal 1?

Reply:
Ooops. That slipped through. Should be 2:

```
In [6]: for i in range(-1, 10):
    print((2 ** i + random.random()) / 4.0)
   ...:     
0.185900067591
0.259383624881
0.666199774509
1.17092891973
2.0800915192
4.21767195389
8.14190244716
16.2083223142
32.1969619394
64.0182664696
128.177728188
```

I'll get a new PR posted to fix this.